File: LeetCode/442. Find All Duplicates in an Array/Solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        result = []
        for i in range(len(nums)):
            x = abs(nums[i])
            idx = x - 1
            if nums[idx] < 0:
                result.append(x)
            else:
                nums[idx] = -nums[idx]
        return result

File: LeetCode/442. Find All Duplicates in an Array/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution{
    public: 
        vector<int> findDuplicates(vector<int>& nums){
            vector<int> result;
            for(int i = 0; i < nums.size(); ++i){
                int x = abs(nums[i]);
                int idx = x - 1;
                if(nums[idx] < 0) {
                    result.push_back(x);
                } else {
                    nums[idx] = -nums[idx];
                }
            }
            return result;
        }
}